### ISSUE TITLE
Gamepad triggers don't seem to be working correctly

### ISSUE THREAD

**Issue by Bingo675** [enigma-dev/enigma-dev#1035]
I'm trying to use gamepad_button_value(0,gp_shoulderrb) for the triggers on a gamepad, but it appears to be stuck at -1.

**Comment by Bingo675** [comment]
After updating the gamepad code from [here](https://github.com/enigma-dev/enigma-dev/pull/966/files#r62594295), my issue still persists, though the value seems to be stuck at 0 now.

Can anyone provide me with a solution, or  practical workaround? I would like to be able to use the triggers.

**Comment by faissaloo** [comment]
0 is only returned when neither the left nor the right trigger is pressed. Try going to line 245 of ENIGMAsystem/SHELL/Universal_System/Extensions/XInput/XIgamepad.cpp and adding:  
```
if (button && button!=1167)
printf("Key pressed: %i\n",button);
else
printf("Error %i\n",button);
```  
before the break and telling us what comes up in a terminal when you run the program (not the ENIGMA console). Also, what platform are you on?

**Comment by RobertBColton** [comment]
I made a mistake and mapped the wrong constants in `gamepad_button_value`. I have fixed it in https://github.com/enigma-dev/enigma-dev/commit/55e43800fb5874f70149fa222cb455c010b2763b which is part of an updated XInput fixes pull request I am creating.

**Comment by RobertBColton** [comment]
Closing as resolved by #1452.
